src/components/Api.js

```diff
@@ -0,0 +1,82 @@
+class Api {
+  constructor(options) {
+    // constructor body
+    this._headers = options.headers;
+    this.baseUrl = options.baseUrl;
+  }
+
+  _checkResponse(res) {
```

Comment:
It's great to have a function for processing the server response, so the logic is not duplicated